app/src/main/java/org/mozilla/tv/firefox/architecture/KillswitchLinearLayout.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.architecture
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.LinearLayout
+import org.mozilla.tv.firefox.components.locale.LocaleManager
+import java.util.Locale
+
+sealed class KillswitchLocales {
+    object All : KillswitchLocales()
+    class ActiveIn(vararg val locales: Locale) : KillswitchLocales()
+}
+
+/**
+ * Can be used to wrap some other [View] in order to prevent it from ever being shown if
+ * certain locale and experiment requirements are not met.
+ *
+ * This is done to prevent bugs similar to #2133. In that bug, a view was set to visibility
+ * GONE when disabled, however its animation toggled that visibility back to VISIBLE.
+ * [KillswitchLinearLayout] allows us to be sure that something will not be displayed if
+ * requirements are not set by intercepting [setVisibility] calls.
+ */
+class KillswitchLinearLayout : LinearLayout {
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
+
+    private var desiredVisibility = this.visibility
```

Comment:
Without holding on to we would have an edge case:
-client code sets visibility to VISIBLE
-> requirements have not yet been set
-> visibility is set to GONE
-client code sets passing requirements

Expected: visibility is VISIBLE
Actual: visibility is GONE

Conceptually, `desiredVisibility` is whatever client code has set, and `visibility` is what has actually happened.  I think it's a fairly clear distinction, but maybe it could use a new name?

Reply:
The comment is helpful but I wish we could better encapsulate that in the name. I can't think of a better one but maybe something jargony for a pending operation like transactions? `pendingVisibility`?